application/dash_application/views/preqc_rna.py

```diff
@@ -0,0 +1,531 @@
+    rna_df = util.df_with_normalized_ius_columns(rna_df, RNA_COL.Run,
+                                                 RNA_COL.Lane, RNA_COL.Barcodes)
+
+    # Calculate percent uniq reads column
+    rna_df[special_cols["Percent Uniq Reads"]] = round(
+        (rna_df[RNA_COL.UniqReads] / rna_df[RNA_COL.TotalReads]) * 100, 1)
+    rna_df[special_cols["Total Reads (Passed Filter)"]] = round(
+        rna_df[RNA_COL.TotalReads] / pow(10, 6), 3)
+    rna_df[special_cols["Percent Correct Strand Reads"]] = round(
+        rna_df[RNA_COL.CorrectStrandReads] * 100
+    )
+
+    # Pull in sample metadata from Pinery. Keep only the columns we care about
+    pinery_samples = util.get_pinery_samples_from_active_projects()
+    pinery_samples = pinery_samples[
+        [
+            "DV200", # FIXME
+            PINERY_COL.IUSTag,
+            PINERY_COL.LaneNumber,
+            PINERY_COL.LibrarySourceTemplateType,
+            PINERY_COL.PrepKit,
+            "RIN", # FIXME
+            PINERY_COL.SampleName,
+            PINERY_COL.SequencerRunName,
+            PINERY_COL.StudyTitle,
+            PINERY_COL.TissueOrigin,
+            PINERY_COL.TissuePreparation,
+            PINERY_COL.TissueType,
+            PINERY_COL.GroupID,
+        ]
+    ]
+    # Filter the Pinery samples for only RNA samples.
+    pinery_samples = util.filter_by_library_design(pinery_samples,
+                                                   ["MR", "SM", "TR", "WT"])
+
+    # Join RNAseqQc and Pinery data
+    rna_df = util.df_with_pinery_samples(rna_df, pinery_samples,
+                                         util.rnaseqqc_ius_columns)
+
+    # Join RNAseqQc and instrument model
+    rna_df = util.df_with_instrument_model(rna_df, PINERY_COL.SequencerRunName)
+
+    return rna_df
+
+# Make the RNA dataframe
+RNA_DF = get_rna_data()
+
+# Build lists of attributes for sorting, shaping, and filtering on
+ALL_PROJECTS = RNA_DF[PINERY_COL.StudyTitle].sort_values().unique()
+ALL_KITS = RNA_DF[PINERY_COL.PrepKit].sort_values().unique()
+ALL_INSTRUMENT_MODELS = RNA_DF[INSTRUMENT_COLS.ModelName].sort_values().unique()
+ALL_TISSUE_MATERIALS = RNA_DF[
+    PINERY_COL.TissuePreparation].sort_values().unique()
+ALL_LIBRARY_DESIGNS = RNA_DF[
+    PINERY_COL.LibrarySourceTemplateType].sort_values().unique()
+ALL_RUNS = RNA_DF[RNA_COL.Run].sort_values().unique()[::-1]  # reverse the list
+
+shape_or_colour_values = {
+    PINERY_COL.StudyTitle: ALL_PROJECTS,
+    PINERY_COL.SequencerRunName: ALL_RUNS,
+    PINERY_COL.PrepKit: ALL_KITS,
+    PINERY_COL.TissuePreparation: ALL_TISSUE_MATERIALS,
+    PINERY_COL.LibrarySourceTemplateType: ALL_LIBRARY_DESIGNS
+}
+
+
+def fill_in_shape_col(df: DataFrame, shape_col: str):
+    all_shapes = get_shapes_for_values(shape_or_colour_values[
+                                           shape_col].tolist())
+    # for each row,
+    df[special_cols['shape']] = df.apply(lambda row: all_shapes.get(row[
+        shape_col]), axis=1)
+    return df
```

Comment:
would it be possible for shape functionality move to plot_builder or df_manipulation? that way scattergl() and scatter_graph() might not have to replicate the generate() in plot_builder as well

Reply:
Yes, that makes sense.

Reply:
(Back story: I developed the scattergl() and scatter_graph() functions while you were working on the plot_builder, and haven't had the time to reimplement them using the plot_builder yet. It's also high on my to-do list)